### PR TITLE
feat: SDK build for 22.0.

### PR DESCRIPTION
**Pull request by drstrangelooker**

### CODE REVIEW

**Review by joeldodge79 — CHANGES_REQUESTED** [review]
I feel pretty strongly that we should cleanup the current state of the past couple releases and release https://github.com/looker-open-source/sdk-codegen/pull/930 as a patch before we merge 22.0 in

File: packages/sdk-codegen/src/python.gen.ts [hunk 1]
```diff
@@ -536,6 +536,7 @@ ${this.hooks.join('\n')}
       number: { default: this.nullStr, name: 'float' },
       password: { default: this.nullStr, name: 'str', asVal: asString },
       string: { default: this.nullStr, name: 'str', asVal: asString },
+      object: { default: this.nullStr, name: 'str', asVal: asString },
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/python.gen.ts:539]: @drstrangelooker what necessitated this change?

**Review by joeldodge79 — CHANGES_REQUESTED** [review]
I reviewed the python generated code changes and it seems good to me. I briefly looked at the workflow file changes and they seem reasonable too though I haven't fully grokked all the changes.

One change we need can be accomplished in two ways.

1. separate out the "update CI for 22.0" changes (workflows etc) from the codegen for 22.0  (generated language files) into a separate PR and merge them first - that seems to be the intention with the title of this PR: `chore: SDK build for 22.0`
2. leave it all combined as is BUT update the title to `feat: Looker 22.0 release` (otherwise release-please will NOT pickup the fact that there are user facing changes in this PR because `chore` is not considered user facing)

**Review by jkaster — COMMENTED** [review]
Agreed with Joel's feedback. Nothing further to add right now.

File: .github/workflows/required-checks-hack-ci.yml [hunk 2]
```diff
@@ -40,30 +40,31 @@ jobs:
...
           filters: |
             apix:
-              - packages/code-editor/**
-              - packages/run-it/**
-              - packages/api-explorer/**
-              - packages/extension-api-explorer/**
-              - packages/extension-utils/**
+              - packages/code-editor/**/*
+              - packages/run-it/**/*
+              - packages/api-explorer/**/*
```
[the remark this answers is not in the document]
Comment (jeremytchang) [reply .github/workflows/required-checks-hack-ci.yml:45]: Is this needed?
Reading their documentation, it doesn't seem necessary? I've tested filters for my Go CI, and `/**` works on subdirectories. 

Also reading [documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths), looks like `/**` is sufficient:

This example runs anytime the push event includes a file in the sub-project directory or its subdirectories, unless the file is in the sub-project/docs directory. For example, a push that changed sub-project/index.js or sub-project/src/index.js will trigger a workflow run, but a push changing only sub-project/docs/readme.md will not.

on:
  push:
    paths:
      - 'sub-project/**'
      - '!sub-project/docs/**'

[the remark this answers is not in the document]
Comment (drstrangelooker) [reply .github/workflows/required-checks-hack-ci.yml:45]: It is needed for the `tony84727/changed-file-filter@v0.2.0`, not for the built in filtering used by the github actions. I logged it as a bug with ` tony84727/changed-file-filter`.

File: packages/sdk-codegen/src/python.gen.ts (hunk 1 above)
Comment (drstrangelooker) [reply packages/sdk-codegen/src/python.gen.ts:539]: There is one place where an object is part of the query string, the `filters` parameter of `"/models/{model_name}/views/{view_name}/fields/{field_name}/suggestions"`. It results in a reference to `mdls.object` which doesn't exist.

**Comment by drstrangelooker** [comment]
@joeldodge79 I renamed it to `feat:`.

**Review by joeldodge79 — APPROVED** [review]
Great work on moving the CI noise-reduction effort forward!!

I'll approve this but with a note: we don't have a good mechanism for testing previous sdk versions against newer Looker versions. And maybe that's not a big deal - if something breaks then upgrade to latest SDK which DO test against all previously supported Looker versions. However, in the old workflow where we modified CI separately and beforehand to test against the next Looker release, we had at least a transient rolling window of "SDK for Looker version N works against Looker version N+1". If the generated code changes for 22.0 are merged together with the CI changes to test against 22.0 then we lose that for SDK(21.20) vs Looker(22.0). But I don't think it's critical, just pointing it out. That and my manta of keeping a PR as single purpose as possible (update CI vs generate new sdk)

**Comment by drstrangelooker** [comment]
OK. I'll do that next time.

**Pull request merged**